diff --git a/src/Protocol/Protocol.php b/src/Protocol/Protocol.php
index 5aa846b..7fe9cbd 100644
--- a/src/Protocol/Protocol.php
+++ b/src/Protocol/Protocol.php
@@ -345,7 +345,6 @@ abstract class Protocol
             self::HEADER_UPGRADE => self::UPGRADE_VALUE,
             self::HEADER_CONNECTION => self::CONNECTION_VALUE,
             self::HEADER_KEY => $key,
-            self::HEADER_ORIGIN => $origin,
             self::HEADER_VERSION => (string) $this->getVersion(),
         ];
     }
